packages/p2p/src/pages/my-profile/block-user/block-user-table/block-user-row/__tests__/block-user-row.spec.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Router } from 'react-router';
+import { createMemoryHistory } from 'history';
+import { useStores } from 'Stores/index';
+import BlockUserRow from '../block-user-row';
+
+const mock_store: DeepPartial<ReturnType<typeof useStores>> = {
+    buy_sell_store: {
+        setSelectedAdState: jest.fn(),
+    },
+    general_store: {
+        is_barred: false,
+    },
+    my_profile_store: {
+        getCounterpartyAdvertiserInfo: jest.fn(),
+        onClickUnblock: jest.fn(),
+    },
+};
+
+const block_user_row_props = {
+    row: {
+        id: '1',
+        is_blocked: 0,
+        name: 'test',
+    },
+};
+
+const history = createMemoryHistory();
```

Comment:
had a doubt.. dont we normally use createbrowserhistory() instead? is there a big difference which will cause any impact? or we just follow it?

Reply:
The `createMemoryHistory` function creates a history object that stores the current location in memory, while `createBrowserHistory` creates a history object that uses the browser's `window.history` API to manage the current location. In the `block-user-row` component, the `createMemoryHistory` function is used to create a history object for testing purposes. If you need to support client-side routing in your application, you can use `createBrowserHistory` instead. However, if you're running your tests in a non-browser environment, using `createBrowserHistory` may not work correctly.